packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -198,23 +186,21 @@ const DatePickerInput = (props) => {
     delete inputAttributes.id;
   }
 
+  let dayInputRef = document.getElementById(dayInputId);
+  let monthInputRef = document.getElementById(monthInputId);
+  let yearInputRef = document.getElementById(yearInputId);
+
   const additionalInputProps = { ...customProps, ...inputAttributes };
   const momentDateFormat = useMemo(() => DateUtil.getFormatByLocale(intl.locale), [intl.locale]);
-  const dateValue = DateUtil.convertToISO8601(value, momentDateFormat);
+  const dateValue = useMemo(() => (DateUtil.convertToISO8601(value, momentDateFormat)), [momentDateFormat, value]);
   const dateFormatOrder = DateUtil.getDateFormatOrder(momentDateFormat);
   const separator = DateUtil.getDateSeparator(intl.locale);
 
-  const setDayRef = useCallback(node => {
-    dayInputRef.current = node;
-  }, []);
-
-  const setMonthRef = useCallback(node => {
-    monthInputRef.current = node;
-  }, []);
-
-  const setYearRef = useCallback(node => {
-    yearInputRef.current = node;
-  }, []);
+  let date = useMemo(() => ({ day: '', month: '', year: '' }), []);
+  // Sets the date state based on the passed in value prop, or if it changes via a calendar click.
+  if (DateUtil.isValidDate(value, momentDateFormat)) {
+    date = DateUtil.getDateInputValues(DateUtil.dateOrder.YMD, dateValue, '-');
```

Comment:
Wouldn't the separator based on the locale instead of using `-`?

Reply:
Never mind, please ignore my comment. `dateValue` is guaranteed to be in the ISO8601 format.